### PR TITLE
format_datetime: using 'S...S' format is incorrect

### PR DESCRIPTION
**Pull request by iamshubh22**
Fixes https://github.com/python-babel/babel/issues/350

### CODE REVIEW

**Comment by akx** [comment]
This is missing test cases.

You would need to prove that the earlier behavior is wrong and that this patch fixes it. :)

**Comment by iamshubh22** [comment]
@akx Oh thanks!! I will see to it. :smile:

**Comment by akx** [comment]
Hey @iamshubh22, okay, so now you have a test case, but I'd still like to see some sort of reasoning (maybe in the form of a reference to the particular piece of specification?) that explains why this new behavior is correct :)

In addition, a couple more test cases wouldn't go amiss, explaining the different cases that the newly implemented behavior covers.

File: babel/dates.py [hunk 1]
```diff
@@ -1287,8 +1287,20 @@ def format_period(self, char):
         return get_period_names(locale=self.locale)[period]
 
     def format_frac_seconds(self, num):
-        value = str(self.value.microsecond)
-        return self.format(round(float('.%s' % value), num) * 10**num, num)
+        value = self.value.microsecond / 1000000
+        return self.format(round(value, num) * 10**num, num)
+        """ Returns fractional seconds. Converts microseconds received to the \
```
Comment (akx) [inline babel/dates.py]: The docstring for a function needs to be _before_ the code, not after. :)

File: babel/dates.py [hunk 2]
```diff
@@ -1287,8 +1287,20 @@ def format_period(self, char):
...
+            nearest round-off value according to the given precesion value.
```
Comment (akx) [inline babel/dates.py]: Typo: it's `precision` :)

File: babel/dates.py [hunk 3]
```diff
@@ -1287,8 +1287,20 @@ def format_period(self, char):
...
+
+            Earlier the function computes wrong result if the microseconds \
```
Comment (akx) [inline babel/dates.py]: This doesn't need to be in the docstring. We don't need to document that a function used to do the wrong thing.

File: babel/dates.py [hunk 4]
```diff
@@ -1287,8 +1287,20 @@ def format_period(self, char):
...
+            received are less than 5 digits.
+
+            d = datetime(2016, 2, 15, 8, 3, 1, 799)
```
Comment (akx) [inline babel/dates.py]: If you are writing doctests, they should be

```pydoc
>>> foo()
result-of-foo
```
, not
```pydoc
foo()
>>> result-of-foo
```

However these shouldn't probably be here either.

File: babel/dates.py (hunk 3 above)
Comment (iamshubh22) [inline babel/dates.py]: @akx So should I remove this docstring part??

Comment (akx) [inline babel/dates.py]: The function should look something like

```python
    def format_frac_seconds(self, num):
        """ Return fractional seconds.

        Rounds the time's microseconds to the precision
        given by the number of digits passed in.
        """
        value = self.value.microsecond / 1000000
        return self.format(round(value, num) * 10**num, num)
```

or similar.

**Pull request closed without merging**